test/jdk/sun/security/pkcs11/SecretKeyFactory/TestGeneral.java

```diff
@@ -99,21 +104,25 @@ public void main(Provider p) throws Exception {
         SecretKey bf_128Key = new SecretKeySpec(rawBytes, 0, 16, "Blowfish");
         SecretKey cc20Key = new SecretKeySpec(rawBytes, 0, 32, "ChaCha20");
 
-        // fixed key length
-        test("AES", aes_128Key, p, TestResult.PASS);
-        test("AES", aes_256Key, p, TestResult.PASS);
-        test("AES", cc20Key, p, TestResult.FAIL);
+        try {
+            // fixed key length
+            test("AES", aes_128Key, p, TestResult.PASS);
+            test("AES", aes_256Key, p, TestResult.PASS);
+            test("AES", cc20Key, p, TestResult.FAIL);
 
-        test("ChaCha20", aes_128Key, p, TestResult.FAIL);
-        test("ChaCha20", aes_256Key, p, TestResult.FAIL);
-        test("ChaCha20", cc20Key, p, TestResult.PASS);
+            test("ChaCha20", aes_128Key, p, TestResult.FAIL);
+            test("ChaCha20", aes_256Key, p, TestResult.FAIL);
+            test("ChaCha20", cc20Key, p, TestResult.PASS);
 
-        // variable key length
-        // Different PKCS11 impls may have different ranges
-        // of supported key sizes for variable-key-length
-        // algorithms.
-        test("Blowfish", aes_128Key, p, TestResult.FAIL);
-        test("Blowfish", cc20Key, p, TestResult.FAIL);
-        test("Blowfish", bf_128Key, p, TestResult.PASS);
+            // variable key length
+            // Different PKCS11 impls may have different ranges
+            // of supported key sizes for variable-key-length
+            // algorithms.
+            test("Blowfish", aes_128Key, p, TestResult.FAIL);
+            test("Blowfish", cc20Key, p, TestResult.FAIL);
+            test("Blowfish", bf_128Key, p, TestResult.PASS);
+        } catch (SkippedException skippedException){
+            throw new SkippedException("One or more tests skipped");
```

Comment:
The original test runs all cases but the updated would stop once any of them is skipped. Is this intended?

Also, I cannot see a benefit to rethrow a `SkippedException` with only a modified message. Why not just let it bubble up?